In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
from sklearn import linear_model
from sklearn.preprocessing import StandardScaler
import warnings
warnings.filterwarnings('ignore')
%matplotlib inline

### Soru 2

In [2]:
sicaklik=pd.read_csv('../weatherHistory.csv')
sicaklik.head()

Formatted Date        Summary Precip Type  Temperature (C)  \
0  2006-04-01 00:00:00.000 +0200  Partly Cloudy        rain         9.472222   
1  2006-04-01 01:00:00.000 +0200  Partly Cloudy        rain         9.355556   
2  2006-04-01 02:00:00.000 +0200  Mostly Cloudy        rain         9.377778   
3  2006-04-01 03:00:00.000 +0200  Partly Cloudy        rain         8.288889   
4  2006-04-01 04:00:00.000 +0200  Mostly Cloudy        rain         8.755556   

   Apparent Temperature (C)  Humidity  Wind Speed (km/h)  \
0                  7.388889      0.89            14.1197   
1                  7.227778      0.86            14.2646   
2                  9.377778      0.89             3.9284   
3                  5.944444      0.83            14.1036   
4                  6.977778      0.83            11.0446   

   Wind Bearing (degrees)  Visibility (km)  Loud Cover  Pressure (millibars)  \
0                   251.0          15.8263         0.0               1015.13   
1                   259.0          15.8263         0.0               1015.63   
2                   204.0          14.9569         0.0               1015.94   
3                   269.0          15.8263         0.0               1016.41   
4                   259.0          15.8263         0.0               1016.51   

                       Daily Summary  
0  Partly cloudy throughout the day.  
1  Partly cloudy throughout the day.  
2  Partly cloudy throughout the day.  
3  Partly cloudy throughout the day.  
4  Partly cloudy throughout the day.

In [3]:
y=sicaklik['Apparent Temperature (C)']-sicaklik['Temperature (C)']
x=sicaklik[['Humidity','Wind Speed (km/h)']]

In [4]:
lrm=linear_model.LinearRegression()
lrm.fit(x,y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [5]:
print('Degiskenler: \n', lrm.coef_)
print('Sabit Deger (bias): \n',lrm.intercept_)

Degiskenler: 
 [-3.02918594 -0.11929075]
Sabit Deger (bias): 
 2.438105415187715


In [6]:
x=sm.add_constant(x)
results=sm.OLS(y,x).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.288
Model:                            OLS   Adj. R-squared:                  0.288
Method:                 Least Squares   F-statistic:                 1.949e+04
Date:                Fri, 03 Jan 2020   Prob (F-statistic):               0.00
Time:                        10:23:21   Log-Likelihood:            -1.7046e+05
No. Observations:               96453   AIC:                         3.409e+05
Df Residuals:                   96450   BIC:                         3.409e+05
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                 2.4381      0.021    115.948      0.000       2.397       2.479
Humidity             -3.0292      0.024   -126.479      0.000      -3.076      -2.982
Wind Speed (km/h)    -0.1193      0.001   -176.164      0.000      -0.121      -0.118
==============================================================================
Omnibus:                     3935.747   Durbin-Watson:                   0.264
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             4613.311
Skew:                          -0.478   Prob(JB):                         0.00
Kurtosis:                       3.484   Cond. No.                         88.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

#### y=2.4381 - 3.0292 x Humidity - 0.1193 x Wind Speed (km/h)

Katsayılar istatistiksel olarak anlamlıdır.Çünkü; p değerleri 0.05 den küçüktür. Hissedilen sıcaklık nem arttıkça artması gerekirken Humidity ile hissedilen sıcaklık ve gerçek sıcaklık farkı arasında negatif bir ilişki vardır. Bunu şurdan anlıyoruz çünkü; Humidity' nin katsayısı -3.0292 dir. Rüzgar hızının etkisi hava sıcaklığı ve hız değerine göre değiştiği için anlamlı bir yorum yapamıyorum.Ancak tahmini katsayılar önceki beklentilerim doğrultusunda değil.

R-squared değeri Humidity ve Wind Speed (km/h) bağımsız değişkenleri ile hedef(bağımlı) değişkeni arasındaki korelasyon değerinin karesini göstermektedir.0.288 değerindeki R-squared bize hedef değişkeni içindeki varyasyonun (çeşitliliğin) %28.8’inin  Humidity ve Wind Speed (km/h) değişkenlerine atfedilebileceğini söylemektedir.

Nem miktarındaki 1 birimlik artış hissedilen sıcaklık ile gerçek sıcaklık farkını 3.0292 (C) azaltacaktır. Bu ilişki anlamlıdır(p_value<0.05).p değeri sıfıra eşittir.
Rüzgar hızındaki 1 birimlik artış ise hissedilen sıcaklık ile gerçek sıcaklık farkını 0.1193 (C) azaltacaktır. Bu ilişki anlamlıdır(p_value<0.05).p değeri sıfıra eşittir.
Hedef ve açıklayıcı değişkenler arasında negatif ilişki vardır.

In [7]:
y=sicaklik['Apparent Temperature (C)']-sicaklik['Temperature (C)']
sicaklik['hum_x_wind']=sicaklik['Humidity']*sicaklik['Wind Speed (km/h)']
x=sicaklik[['Humidity','Wind Speed (km/h)','hum_x_wind']]

In [8]:
lrm=linear_model.LinearRegression()
lrm.fit(x,y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [9]:
print('Degiskenler: \n', lrm.coef_)
print('Sabit Deger (bias): \n',lrm.intercept_)

Degiskenler: 
 [ 0.17751219  0.09048213 -0.29711946]
Sabit Deger (bias): 
 0.08393631009784941


In [10]:
x=sm.add_constant(x)
results=sm.OLS(y,x).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.341
Model:                            OLS   Adj. R-squared:                  0.341
Method:                 Least Squares   F-statistic:                 1.666e+04
Date:                Fri, 03 Jan 2020   Prob (F-statistic):               0.00
Time:                        10:23:24   Log-Likelihood:            -1.6669e+05
No. Observations:               96453   AIC:                         3.334e+05
Df Residuals:                   96449   BIC:                         3.334e+05
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                 0.0839      0.033      2.511      0.012       0.018       0.149
Humidity              0.1775      0.043      4.133      0.000       0.093       0.262
Wind Speed (km/h)     0.0905      0.002     36.797      0.000       0.086       0.095
hum_x_wind           -0.2971      0.003    -88.470      0.000      -0.304      -0.291
==============================================================================
Omnibus:                     4849.937   Durbin-Watson:                   0.262
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             9295.404
Skew:                          -0.378   Prob(JB):                         0.00
Kurtosis:                       4.320   Cond. No.                         193.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

#### y=0.0839+0.1775 x Humidity + 0.0905 x Wind Speed (km/h) - 0.2971 x hum_x_wind
Katsayılar istatistiksel olarak anlamlı. Katsayılar değişti ve hedef değişken ile açıklayıcı değişkenler(Humidity,Wind Speed (km/h)) arasındaki ilişki pozitif oldu.

### Soru 3

In [11]:
house=pd.read_csv('../trainn.csv')
house.head()

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities  ... PoolArea PoolQC Fence MiscFeature MiscVal MoSold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   
3         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
4         Lvl    AllPub  ...        0    NaN   NaN         NaN       0     12   

  YrSold  SaleType  SaleCondition  SalePrice  
0   2008        WD         Normal     208500  
1   2007        WD         Normal     181500  
2   2008        WD         Normal     223500  
3   2006        WD        Abnorml     140000  
4   2008        WD         Normal     250000  

[5 rows x 81 columns]

Ön veri araştırması sonucu hedef değişkenimizi tahmin etmek için kullanacağım bir kaç özellik belirledim.


In [12]:
housee=house[['MSSubClass','Neighborhood','Condition2',
                          'OverallCond','Exterior1st','ExterCond','Heating','SaleType','SalePrice']]
housee.head()

MSSubClass Neighborhood Condition2  OverallCond Exterior1st ExterCond  \
0          60      CollgCr       Norm            5     VinylSd        TA   
1          20      Veenker       Norm            8     MetalSd        TA   
2          60      CollgCr       Norm            5     VinylSd        TA   
3          70      Crawfor       Norm            5     Wd Sdng        TA   
4          60      NoRidge       Norm            5     VinylSd        TA   

  Heating SaleType  SalePrice  
0    GasA       WD     208500  
1    GasA       WD     181500  
2    GasA       WD     223500  
3    GasA       WD     140000  
4    GasA       WD     250000

In [13]:
SubClassDict = {20: "1-STORY 1946 & NEWER ALL STYLES", 
        30: "1-STORY 1945 & OLDER", 
        40: "1-STORY W/FINISHED ATTIC ALL AGES", 
        45: "1-1/2 STORY - UNFINISHED ALL AGES", 
        50: "1-1/2 STORY FINISHED ALL AGES", 
        60: "2-STORY 1946 & NEWER", 
        70: "2-STORY 1945 & OLDER", 
        75: "2-1/2 STORY ALL AGES", 
        80: "SPLIT OR MULTI-LEVEL", 
        85: "SPLIT FOYER", 
        90: "DUPLEX - ALL STYLES AND AGES", 
       120: "1-STORY PUD (Planned Unit Development) - 1946 & NEWER", 
       150: "1-1/2 STORY PUD - ALL AGES", 
       160: "2-STORY PUD - 1946 & NEWER", 
       180: "PUD - MULTILEVEL - INCL SPLIT LEV/FOYER", 
       190: "2 FAMILY CONVERSION - ALL STYLES AND AGES"}

In [14]:
OverallCondDict={10:"Very Excellent",
       9:"Excellent",
       8:"Very Good",
       7:"Good",
       6:"Above Average",
       5:"Average",
       4:"Below Average",
       3:"Fair",
       2:"Poor",
       1:"Very Poor"}

In [15]:
housee.MSSubClass = housee.MSSubClass.astype('category')
housee.OverallCond= housee.OverallCond.astype('category')

In [16]:
housee.MSSubClass.map(SubClassDict).head()

0               2-STORY 1946 & NEWER
1    1-STORY 1946 & NEWER ALL STYLES
2               2-STORY 1946 & NEWER
3               2-STORY 1945 & OLDER
4               2-STORY 1946 & NEWER
Name: MSSubClass, dtype: object

In [17]:
housee.OverallCond.map(OverallCondDict).head()

0      Average
1    Very Good
2      Average
3      Average
4      Average
Name: OverallCond, dtype: object

In [18]:
endocing=pd.concat([housee,pd.get_dummies(housee[['Neighborhood','Condition2',
                                        'Exterior1st','ExterCond','Heating','SaleType','MSSubClass','OverallCond']])],axis=1).drop(columns=['Neighborhood','Condition2',
                                        'Exterior1st','ExterCond','Heating','SaleType','MSSubClass','OverallCond'],axis=1)

In [19]:
y=endocing['SalePrice']
x=pd.get_dummies(housee[['Neighborhood','Condition2',
                                        'Exterior1st','ExterCond','Heating','SaleType','MSSubClass','OverallCond']])

In [20]:
import statsmodels.api as sm
pd.options.display.float_format='{:,.7f}'.format
x = sm.add_constant(x)
results = sm.OLS(y, x).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              SalePrice   R-squared:                       0.667
Model:                            OLS   Adj. R-squared:                  0.647
Method:                 Least Squares   F-statistic:                     32.85
Date:                Fri, 03 Jan 2020   Prob (F-statistic):          2.71e-269
Time:                        10:23:32   Log-Likelihood:                -17740.
No. Observations:                1460   AIC:                         3.565e+04
Df Residuals:                    1375   BIC:                         3.610e+04
Df Model:                          84                                         
Covariance Type:            nonrobust                                         
========================================================================================
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                 8.255e+04   1.25e+04      6.595      0.000     5.8e+04    1.07e+05
Neighborhood_Blmngtn   4.23e+04   1.27e+04      3.320      0.001    1.73e+04    6.73e+04
Neighborhood_Blueste  1.032e+04   3.34e+04      0.309      0.757   -5.52e+04    7.58e+04
Neighborhood_BrDale  -2.127e+04   1.41e+04     -1.505      0.133    -4.9e+04    6454.795
Neighborhood_BrkSide -4.023e+04   7973.481     -5.045      0.000   -5.59e+04   -2.46e+04
Neighborhood_ClearCr  1.949e+04   9259.893      2.105      0.035    1328.333    3.77e+04
Neighborhood_CollgCr   108.4102   5116.060      0.021      0.983   -9927.717    1.01e+04
Neighborhood_Crawfor  1.393e+04   7778.644      1.791      0.073   -1326.519    2.92e+04
Neighborhood_Edwards -4.477e+04   5524.119     -8.105      0.000   -5.56e+04   -3.39e+04
Neighborhood_Gilbert -1.791e+04   6444.122     -2.780      0.006   -3.06e+04   -5272.165
Neighborhood_IDOTRR   -6.72e+04   9031.010     -7.441      0.000   -8.49e+04   -4.95e+04
Neighborhood_MeadowV -7.936e+04   1.62e+04     -4.901      0.000   -1.11e+05   -4.76e+04
Neighborhood_Mitchel -2.292e+04   7296.058     -3.141      0.002   -3.72e+04   -8603.560
Neighborhood_NAmes   -4.199e+04   4453.853     -9.428      0.000   -5.07e+04   -3.33e+04
Neighborhood_NPkVill   413.0911   1.67e+04      0.025      0.980   -3.23e+04    3.31e+04
Neighborhood_NWAmes  -8701.1979   6377.149     -1.364      0.173   -2.12e+04    3808.797
Neighborhood_NoRidge  1.306e+05   8147.964     16.031      0.000    1.15e+05    1.47e+05
Neighborhood_NridgHt  1.172e+05   6406.778     18.300      0.000    1.05e+05     1.3e+05
Neighborhood_OldTown -5.566e+04   6267.736     -8.881      0.000    -6.8e+04   -4.34e+04
Neighborhood_SWISU   -4.078e+04   1.05e+04     -3.887      0.000   -6.14e+04   -2.02e+04
Neighborhood_Sawyer  -4.608e+04   6274.131     -7.344      0.000   -5.84e+04   -3.38e+04
Neighborhood_SawyerW    18.1065   6727.597      0.003      0.998   -1.32e+04    1.32e+04
Neighborhood_Somerst  2.727e+04   6098.244      4.473      0.000    1.53e+04    3.92e+04
Neighborhood_StoneBr  1.236e+05   9980.846     12.388      0.000    1.04e+05    1.43e+05
Neighborhood_Timber   4.169e+04   8101.911      5.146      0.000    2.58e+04    5.76e+04
Neighborhood_Veenker  4.236e+04   1.46e+04      2.897      0.004    1.37e+04    7.11e+04
Condition2_Artery    -1.561e+04   3.33e+04     -0.468      0.640    -8.1e+04    4.98e+04
Condition2_Feedr     -2.259e+04   2.29e+04     -0.988      0.323   -6.74e+04    2.23e+04
Condition2_Norm      -1717.5406   1.31e+04     -0.131      0.896   -2.74e+04     2.4e+04
Condition2_PosA       1.482e+05   5.34e+04      2.775      0.006    4.34e+04    2.53e+05
Condition2_PosN      -3.315e+04   3.31e+04     -1.002      0.316    -9.8e+04    3.17e+04
Condition2_RRAe       5.498e+04   4.41e+04      1.247    

In [32]:
results.pvalues[results.pvalues<0.05].index.values

array(['const', 'Neighborhood_Blmngtn', 'Neighborhood_BrkSide',
       'Neighborhood_ClearCr', 'Neighborhood_Edwards',
       'Neighborhood_Gilbert', 'Neighborhood_IDOTRR',
       'Neighborhood_MeadowV', 'Neighborhood_Mitchel',
       'Neighborhood_NAmes', 'Neighborhood_NoRidge',
       'Neighborhood_NridgHt', 'Neighborhood_OldTown',
       'Neighborhood_SWISU', 'Neighborhood_Sawyer',
       'Neighborhood_Somerst', 'Neighborhood_StoneBr',
       'Neighborhood_Timber', 'Neighborhood_Veenker', 'Condition2_PosA',
       'Exterior1st_BrkFace', 'Exterior1st_CemntBd', 'Exterior1st_Stucco',
       'ExterCond_Gd', 'ExterCond_TA', 'Heating_GasA', 'Heating_GasW',
       'SaleType_New', 'MSSubClass_20', 'MSSubClass_30', 'MSSubClass_50',
       'MSSubClass_60', 'MSSubClass_70', 'MSSubClass_75', 'MSSubClass_80',
       'MSSubClass_120', 'MSSubClass_160', 'OverallCond_9'], dtype=object)

In [22]:
anlamlilar=pd.concat([endocing.SalePrice,endocing[results.pvalues[results.pvalues<0.05].index.drop('const')]],axis=1)

In [23]:
y=anlamlilar['SalePrice']
x=anlamlilar.drop(columns='SalePrice')
import statsmodels.api as sm
x=sm.add_constant(x)
result=sm.OLS(y,x).fit()
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              SalePrice   R-squared:                       0.649
Model:                            OLS   Adj. R-squared:                  0.640
Method:                 Least Squares   F-statistic:                     71.22
Date:                Fri, 03 Jan 2020   Prob (F-statistic):          6.81e-293
Time:                        10:23:38   Log-Likelihood:                -17779.
No. Observations:                1460   AIC:                         3.563e+04
Df Residuals:                    1422   BIC:                         3.583e+04
Df Model:                          37                                         
Covariance Type:            nonrobust                                         
========================================================================================
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                 9.939e+04   1.58e+04      6.298      0.000    6.84e+04     1.3e+05
Neighborhood_Blmngtn  4.175e+04   1.31e+04      3.176      0.002     1.6e+04    6.75e+04
Neighborhood_BrkSide -4.686e+04   7736.865     -6.057      0.000    -6.2e+04   -3.17e+04
Neighborhood_ClearCr  2.212e+04   9435.436      2.345      0.019    3613.660    4.06e+04
Neighborhood_Edwards -5.031e+04   5715.420     -8.803      0.000   -6.15e+04   -3.91e+04
Neighborhood_Gilbert -1.574e+04   6223.140     -2.530      0.012    -2.8e+04   -3535.246
Neighborhood_IDOTRR  -7.191e+04   9163.657     -7.848      0.000   -8.99e+04   -5.39e+04
Neighborhood_MeadowV -9.375e+04   1.45e+04     -6.478      0.000   -1.22e+05   -6.54e+04
Neighborhood_Mitchel -2.611e+04   7408.317     -3.525      0.000   -4.06e+04   -1.16e+04
Neighborhood_NAmes   -4.361e+04   4281.113    -10.187      0.000    -5.2e+04   -3.52e+04
Neighborhood_NoRidge  1.295e+05   8089.219     16.015      0.000    1.14e+05    1.45e+05
Neighborhood_NridgHt  1.192e+05   6420.149     18.569      0.000    1.07e+05    1.32e+05
Neighborhood_OldTown -5.862e+04   6156.036     -9.522      0.000   -7.07e+04   -4.65e+04
Neighborhood_SWISU   -4.287e+04   1.07e+04     -4.017      0.000   -6.38e+04   -2.19e+04
Neighborhood_Sawyer   -4.81e+04   6244.765     -7.703      0.000   -6.04e+04   -3.59e+04
Neighborhood_Somerst  3.262e+04   6181.974      5.276      0.000    2.05e+04    4.47e+04
Neighborhood_StoneBr  1.209e+05   1.04e+04     11.617      0.000    1.01e+05    1.41e+05
Neighborhood_Timber   4.345e+04   8218.306      5.288      0.000    2.73e+04    5.96e+04
Neighborhood_Veenker  5.099e+04   1.48e+04      3.440      0.001    2.19e+04    8.01e+04
Condition2_PosA       1.388e+05    5.1e+04      2.722      0.007    3.88e+04    2.39e+05
Exterior1st_BrkFace   3.848e+04   7105.187      5.416      0.000    2.45e+04    5.24e+04
Exterior1st_CemntBd   4.698e+04   7678.920      6.118      0.000    3.19e+04     6.2e+04
Exterior1st_Stucco      2.7e+04   1.02e+04      2.658      0.008    7072.587    4.69e+04
ExterCond_Gd          3.368e+04   9791.355      3.440      0.001    1.45e+04    5.29e+04
ExterCond_TA          2.643e+04   9119.688      2.898      0.004    8536.284    4.43e+04
Heating_GasA          4.374e+04   1.31e+04      3.341      0.001    1.81e+04    6.94e+04
Heating_GasW            5.6e+04   1.72e+04      3.255      0.001    2.23e+04    8.98e+04
SaleType_New          3.386e+04   5186.592      6.528      0.000    2.37e+04     4.4e+04
MSSubClass_20         1.498e+04   5022.408      2.983      0.003    5132.134    2.48e+04
MSSubClass_30        -2.244e+04   7657.455     -2.930      0.003   -3.75e+04   -7415.688
MSSubClass_50         1.099e+04   6112.246      1.797      0.072   -1003.236     2.3e+04
MSSubClass_60         3.976e+04   5781.506      6.877    

Anlamli olan özellikler ile çalıştığımda R-squared'im 0.64 oldu.İlk duruma göre azaldı.İyileştirme olmadı.

Aşağıda anlamli olan özelliklerin bir birim değiştiğinde SalePrice'ı kaç birim etkileyeceğini belirlediğimiz bu özelliklere ait katsayıları görüyoruz.

In [24]:
result.params[result.pvalues<0.05]

const                   99,390.0945431
Neighborhood_Blmngtn    41,749.6982147
Neighborhood_BrkSide   -46,860.2746792
Neighborhood_ClearCr    22,122.5284252
Neighborhood_Edwards   -50,313.0115566
Neighborhood_Gilbert   -15,742.7676589
Neighborhood_IDOTRR    -71,914.9343965
Neighborhood_MeadowV   -93,749.1408056
Neighborhood_Mitchel   -26,114.6970947
Neighborhood_NAmes     -43,611.6953401
Neighborhood_NoRidge   129,545.0729385
Neighborhood_NridgHt   119,215.5937044
Neighborhood_OldTown   -58,615.6846508
Neighborhood_SWISU     -42,870.8325032
Neighborhood_Sawyer    -48,102.0650190
Neighborhood_Somerst    32,617.2092397
Neighborhood_StoneBr   120,949.8539178
Neighborhood_Timber     43,454.7337938
Neighborhood_Veenker    50,985.3121629
Condition2_PosA        138,809.8246865
Exterior1st_BrkFace     38,482.2450304
Exterior1st_CemntBd     46,982.1238967
Exterior1st_Stucco      27,002.8614319
ExterCond_Gd            33,678.8837774
ExterCond_TA            26,425.7699881
Heating_GasA            4

Yukaridaki ozellikler SalePrice üzerinde belirgin etkiye sahiptir.
Anlamlilar ile çalıştığımda modele giren kısıtlar 87 den 37 ye düştü. Modelim iyileşti. R-squared im %64 e düştü. Düşmesinin nedeni ise p_value 0.05 e yakın aslında anlamlı denebilecek(0.06 gibi sıfıra yakın) değerler vardı. Bu değerleride attığımızda R-squared azaldı ancak çok düşük miktarda. Sonuç mantıklı.